packages/browser/src/login/oidc/redirectHandler/ImplicitRedirectHandler.ts

```diff
@@ -29,41 +29,43 @@ import {
   ISessionInfo,
   ISessionInfoManager,
 } from "@inrupt/solid-client-authn-core";
-import URL from "url-parse";
+import UrlParse from "url-parse";
 import { inject, injectable } from "tsyringe";
 import { ITokenSaver } from "./TokenSaver";
 import { buildBearerFetch } from "../../../authenticatedFetch/fetchFactory";
-import { getUnauthenticatedSession } from "../../../sessionInfo/SessionInfoManager";
 
 /**
  * @hidden
  */
 @injectable()
-export default class GeneralRedirectHandler implements IRedirectHandler {
+export class ImplicitRedirectHandler implements IRedirectHandler {
   constructor(
     @inject("tokenSaver") private tokenSaver: ITokenSaver,
     @inject("sessionInfoManager")
     private sessionInfoManager: ISessionInfoManager
   ) {}
 
   async canHandle(redirectUrl: string): Promise<boolean> {
-    const url = new URL(redirectUrl, true);
-    return !!(
-      url.query &&
-      url.query.id_token &&
-      url.query.access_token &&
-      url.query.state
-    );
+    try {
+      const myUrl = new URL(redirectUrl);
+      return (
+        myUrl.searchParams.get("id_token") !== null &&
+        myUrl.searchParams.get("access_token") !== null &&
+        myUrl.searchParams.get("state") !== null
+      );
+    } catch {
+      throw new Error(
+        `[${redirectUrl}] is not a valid URL, and cannot be used as a redirect URL.`
+      );
+    }
   }
   async handle(
     redirectUrl: string
   ): Promise<ISessionInfo & { fetch: typeof fetch }> {
     if (!(await this.canHandle(redirectUrl))) {
-      // If the received IRI does not have redirection information, we can only
-      // return an unauthenticated session.
-      return getUnauthenticatedSession();
+      throw new Error(`ImplicitRedirectHandler cannot handle [${redirectUrl}]`);
```

Comment:
I'd be more explicit in this message. If I understand correctly, the reason we throw here it that we're assuming we are the last handler in the chain of all handlers - i.e. that we are the handler of last-resort - is that correct? If so, I'd somehow allude to that in this message, as it seems like important logic to understand.

Reply:
To the contrary, we are throwing there because we are **not** the last handler: the newly created `FallbackRedirectHandler` is the one who should get called last, because as long as the URL you give to it is valid, it says it can handle it, and return an unauthenticated session. 

The `ImplicitRedirectHandler` and `AuthCodeRedirectHandler` throwing in their `handle` method is  possible because in the way we implement the handler pattern, `handle` should never be called if `canHandle` returns false. That's why I added a `FallbackRedirectHandler`, so that the redirect handler aggregator would always find a suitable handler, and a user can always use `handleRedirect` on any IRI, without having it throw an error in case the IRI is not a redirect one. In such case, you'd just get an unauthenticated session.

Reply:
This context is (tentatively) given in the `FallbackRedirectHandler` class and in the `dependencies` file, where the handler is registered.